compact_str/src/repr/static_str.rs

```diff
@@ -0,0 +1,31 @@
+use core::mem;
+
+use super::{
+    Repr,
+    MAX_SIZE,
+    STATIC_STR_MASK,
+};
+
+pub(super) const DISCRIMINANT_SIZE: usize = MAX_SIZE - mem::size_of::<&'static str>();
+
+/// A buffer stored on the stack whose size is equal to the stack size of `String`
+/// The last byte is set to 0.
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct StaticStr {
+    pub text: &'static str,
+    #[allow(unused)]
+    discriminant: [u8; DISCRIMINANT_SIZE],
+}
```

Comment:
What do you think of "deconstructing" the `&'static str` and changing this repr to
```
pub struct StaticStr {
  pub ptr: ptr::NonNull<u8>,
  pub len: usize,
  discriminant: [u8; DISCRIMINANT_SIZE],
}
```

I'll need to double check some of the guarantees around static memory, but I think this layout should work just fine since the static mem should not change for the lifetime of the program.

Reply:
Agree, that should work.

Reply:
I've applied this and it indeed works!